### PR TITLE
refactor: upgrade Fabric.js to 6.5.1

### PR DESCRIPTION
**Pull request by xfqz86**
refactor: upgrade Fabric.js to 6.5.1
ci: add repository check condition to GitHub workflows

### CODE REVIEW

File: src/fabric-object/barcode.class.ts [hunk 1]
```diff
@@ -1,40 +1,84 @@
-import { fabric } from "fabric";
+import * as fabric from "fabric";
 import { code128b, ean13 } from "../utils/barcode";
 import { equalSpacingFillText } from "../utils/canvas_utils";
 import { OBJECT_DEFAULTS_TEXT } from "../defaults";
 
 const ALL_PROPERTIES = ["text", "encoding", "printText", "scaleFactor", "fontSize", "font", "fontFamily"];
```
Comment (MultiMote) [inline src/fabric-object/barcode.class.ts:6]: Can be replaced by BARCODE_PROPS

File: src/fabric-object/qrcode.class.ts [hunk 2]
```diff
@@ -1,85 +1,100 @@
...
 
-export interface IQRCodeOptions extends fabric.IObjectOptions {
-  text: string;
-  ecl?: ErrorCorrectionLevel;
+export const qrcodeDefaultValues: Partial<fabric.TClassProperties<QRCode>> = {
+  text: "Text",
+  ecl: "M",
+  stroke: "#000000",
+  fill: "#ffffff",
+  ...DEFAULT_LABEL_PROPS.size,
```
Comment (MultiMote) [inline src/fabric-object/qrcode.class.ts:11]: Should be OBJECT_SIZE_DEFAULTS I think (you are using 240x96 currently)

**Comment by MultiMote** [comment]
Barcode becomes blurry every odd character number.

[vivaldi_A6T64K38wS.webm](https://github.com/user-attachments/assets/d08c10ff-a467-4306-ad58-0d38d1e7fd05)

**Review by MultiMote — CHANGES_REQUESTED** [review]

File: src/utils/file_utils.ts [hunk 3]
```diff
@@ -136,18 +137,11 @@ export class FileUtils {
...
-        },
-        (src: object, obj: fabric.Object /*, error: any*/) => {
-          obj.set({ snapAngle: OBJECT_DEFAULTS.snapAngle });
-        }
-      );
+    await canvas.loadFromJSON(state, (_, obj) => {
+      if ('set' in obj)
+        obj.set({ snapAngle: OBJECT_DEFAULTS.snapAngle });
     });
+    canvas.backgroundColor = "#fff";
```
Comment (MultiMote) [inline src/utils/file_utils.ts:144]: Can be removed (custom canvas do is not using this property)

**Comment by xfqz86** [comment]
> Code128 barcode becomes blurry every odd character number for x1 scale.
> vivaldi_A6T64K38wS.webm

I can't reproduce the problem on my computer


https://github.com/user-attachments/assets/6b5c1cf7-d192-4487-9acd-0b05cb690706

**Comment by MultiMote** [comment]
> I can't reproduce the problem on my computer

Maybe it related to screen DPI. I have ordinary 1920x1080 monitor.

In any case, this problem was not observed in the main branch.

I tried overriding shouldCache (`this.objectCaching = false` used before). This can be used as a temporary solution.


[vivaldi_6lovojwWZ7.webm](https://github.com/user-attachments/assets/b47799b6-2aa1-4ecf-bec2-4fa119344150)

**Comment by xfqz86** [comment]
> > I can't reproduce the problem on my computer
> 
> Maybe it related to screen DPI. I have ordinary 1920x1080 monitor.
> 
> In any case, this problem was not observed in the main branch.
> 
> I tried overriding shouldCache (`this.objectCaching = false` used before). This can be used as a temporary solution.
> vivaldi_6lovojwWZ7.webm

I made some changes. See if it works

**Comment by MultiMote** [comment]
> I made some changes. See if it works

Yes, it works.

**Comment by MultiMote** [comment]
Mirroring is broken now

vivaldi_rtvXUVjFrY

vivaldi_Zm5CjvC55M

[label_1732786794.json](https://github.com/user-attachments/files/17946083/label_1732786794.json)

**Comment by MultiMote** [comment]
Fixed by making `CustomCanvas.createMirroredObjects` async

```ts
async createMirroredObjects() {
    const objects = this.getObjects();
    for (const obj of objects) {
      const info = this.getMirroredObjectCoords(obj);
      if (info != undefined) {
        const newObj = await obj.clone();
        newObj.setPositionByOrigin(info.pos, "center", "center");
        if (info.flip) {
          newObj.centeredRotation = true;
          newObj.rotate(180);
        }
        this.add(newObj);
      }
    }
  }
```

**Comment by xfqz86** [comment]
> Can be fixed by making `CustomCanvas.createMirroredObjects` async
> 
> ```ts
> async createMirroredObjects() {
>     const objects = this.getObjects();
>     for (const obj of objects) {
>       const info = this.getMirroredObjectCoords(obj);
>       if (info != undefined) {
>         const newObj = await obj.clone();
>         newObj.setPositionByOrigin(info.pos, "center", "center");
>         if (info.flip) {
>           newObj.centeredRotation = true;
>           newObj.rotate(180);
>         }
>         this.add(newObj);
>       }
>     }
>   }
> ```

You are right. I submit this fix

File: src/utils/image_editor_utils.ts
Comment (MultiMote) [inline src/utils/image_editor_utils.ts:1]: Exclamation marks can be removed now.

**Review by MultiMote — APPROVED** [review]
I'm going to merge this into the temporary branch and do some tweaking myself.

**Pull request merged**